hail/src/main/scala/is/hail/expr/ir/functions/RelationalFunctions.scala

```diff
@@ -25,6 +27,44 @@ abstract class MatrixToTableFunction {
   def execute(mv: MatrixValue): TableValue
 
   def preservesPartitionCounts: Boolean
+
+  def lower(): Option[TableToTableFunction] = None
+}
+
+case class WrappedMatrixToTableFunction(
+  function: MatrixToTableFunction,
+  colsFieldName: String,
+  entriesFieldName: String,
+  colKey: IndexedSeq[String]) extends TableToTableFunction {
+  override def typeInfo(childType: TableType, childRVDType: RVDType): (TableType, RVDType) = {
+    val mType = MatrixType.fromTableType(childType, colsFieldName, entriesFieldName, colKey)
+    function.typeInfo(mType, mType.canonicalRVDType) // MatrixType RVDTypes will go away
+  }
+
+  def execute(tv: TableValue): TableValue = function.execute(tv.toMatrixValue(colsFieldName, entriesFieldName, colKey))
+
+  override def preservesPartitionCounts: Boolean = function.preservesPartitionCounts
+}
+
+case class WrappedMatrixToMatrixFunction(function: MatrixToMatrixFunction,
+  inColsFieldName: String,
+  outColsFieldName: String,
+  inEntriesFieldName: String,
```

Comment:
Are these in/out field names usually always the same?

Reply:
I think yes, but it looks like the lowerer supports that not being the case. Better safe than sorry.